File: python-client-sdk/src/sdk/models/shared/codat_datacontracts_datasets_bankstatementlinepagedresponsemodel.py
from __future__ import annotations
import dataclasses
from ..shared import codat_datacontracts_datasets_bankstatementline as shared_codat_datacontracts_datasets_bankstatementline
from ..shared import codat_datacontracts_datasets_bankstatementlinepagedresponselinksmodel as shared_codat_datacontracts_datasets_bankstatementlinepagedresponselinksmodel
from dataclasses_json import Undefined, dataclass_json
from sdk import utils
from typing import Optional


@dataclass_json(undefined=Undefined.EXCLUDE)
@dataclasses.dataclass
class CodatDataContractsDatasetsBankStatementLinePagedResponseModel:
    r"""CodatDataContractsDatasetsBankStatementLinePagedResponseModel
    Used to represent what can be returned by an endpoint that supports paging.
    Usable with the [ProducesResponseType] attribute on a controller action.
    """
    
    links: Optional[shared_codat_datacontracts_datasets_bankstatementlinepagedresponselinksmodel.CodatDataContractsDatasetsBankStatementLinePagedResponseLinksModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('_links'), 'exclude': lambda f: f is None }})
    page_number: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('pageNumber'), 'exclude': lambda f: f is None }})
    page_size: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('pageSize'), 'exclude': lambda f: f is None }})
    results: Optional[list[shared_codat_datacontracts_datasets_bankstatementline.CodatDataContractsDatasetsBankStatementLine]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('results'), 'exclude': lambda f: f is None }})
    total_results: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('totalResults'), 'exclude': lambda f: f is None }})
    

File: go-client-sdk/pkg/models/shared/codatdatacontractsdatasetsbankstatementlinepagedresponsemodel.go
package shared

// CodatDataContractsDatasetsBankStatementLinePagedResponseModel
// Used to represent what can be returned by an endpoint that supports paging.
// Usable with the [ProducesResponseType] attribute on a controller action.
type CodatDataContractsDatasetsBankStatementLinePagedResponseModel struct {
	Links        *CodatDataContractsDatasetsBankStatementLinePagedResponseLinksModel `json:"_links,omitempty"`
	PageNumber   *int                                                                `json:"pageNumber,omitempty"`
	PageSize     *int                                                                `json:"pageSize,omitempty"`
	Results      []CodatDataContractsDatasetsBankStatementLine                       `json:"results,omitempty"`
	TotalResults *int                                                                `json:"totalResults,omitempty"`
}
